tests/unit/backend/helpers/generate-form-data.ts

```diff
@@ -117,12 +118,35 @@ export const generateSingleAnswerResponse = (
     answer,
     fieldType: field.fieldType as Exclude<
       BasicField,
-      BasicField.Table | BasicField.Checkbox
+      BasicField.Table | BasicField.Checkbox | BasicField.Attachment
     >,
     isVisible: true,
   }
 }
 
+export const generateSingleAnswerResponse = (
+  field: IFieldSchema,
+  answer = 'answer',
+): ISingleAnswerResponse => {
+  if (
+    [BasicField.Attachment, BasicField.Table, BasicField.Checkbox].includes(
+      field.fieldType,
+    )
+  ) {
+    throw new Error(
+      'Call the custom response generator functions for attachment, table and checkbox.',
+    )
+  }
+  return {
+    _id: field._id,
+    answer,
+    fieldType: field.fieldType as Exclude<
+      BasicField,
+      BasicField.Table | BasicField.Checkbox | BasicField.Attachment
+    >,
```

Comment:
this should include `question` right?

Reply:
nope, because question is only injected to ProcessedResponse; ISingleAnswerResponse (or FieldResponse in general) does not have question:

```
export interface IBaseResponse {
  _id: IFieldSchema['_id']
  fieldType: BasicField
  myInfo?: IMyInfo
  // Signature exists for verifiable fields if the answer is verified.
  signature?: string
}

export interface ISingleAnswerResponse extends IBaseResponse {
  fieldType: Exclude<
    BasicField,
    BasicField.Table | BasicField.Checkbox | BasicField.Attachment
  >
  answer: string
}
```

Reply:
^ thanks for highlighting, removed the unnecessary !response.isVisible